connectors/sources/sharepoint.py

```diff
@@ -569,12 +597,12 @@ async def get_list_items(self, list_id, site_url, server_relative_url):
             dictionary: dictionary containing meta-data of the list item.
         """
         file_relative_url = None
-        async for list_items_data in self.invoke_get_call(
+        async for list_items_data in self.sharepoint_client.invoke_get_call(
             site_url=site_url, param_name=LIST_ITEM, list_id=list_id
         ):
             for result in list_items_data:
                 if not result.get("Attachments"):
-                    url = f"{self.host_url}{server_relative_url}/DispForm.aspx?ID={result['Id']}&Source={self.host_url}{server_relative_url}/AllItems.aspx&ContentTypeId={result['ContentTypeId']}"
+                    url = f"{self.sharepoint_client.host_url}{server_relative_url}/DispForm.aspx?ID={result['Id']}&Source={self.sharepoint_client.host_url}{server_relative_url}/AllItems.aspx&ContentTypeId={result['ContentTypeId']}"
```

Comment:
These kinds of templates to build API path strings feel like they belong in the client. Any reason why this whole function shouldn't be in the client too?

Reply:
Here it is not interacting with SharePoint. It just manipulating the URL.